0310-minimum-height-trees.py

```diff
@@ -0,0 +1,57 @@
+from collections import defaultdict, deque
+from typing import List
+
```

Comment:
Adding comments to explain the logic of certain sections can be helpful for better understanding.